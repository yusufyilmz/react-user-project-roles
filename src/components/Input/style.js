
import styled from 'styled-components';

export const Label = styled.label`
    float: left;
    width: 25%;
    margin: 15px 5px;
    font-size: x-large;
    `

Label.displayName = 'Label'


export const Input = styled.input`
    margin-left: auto;
    margin-right: auto;
    width: 100%;
    border: 1px solid #ccc;
    border-radius: 4px;
    resize: vertical;
    background-color: ${props => (props.type === 'submit' || props.type === 'button') && '#27435c'};
    display: ${props => (props.type === 'submit' || props.type === 'button') && 'flex'};
    justify-content: ${props => (props.type === 'submit' || props.type === 'button') && 'center'};
    color: ${props => (props.type === 'submit' || props.type === 'button') && 'white'};
    height: ${props => (props.type === 'submit' || props.type === 'button') && '50px'};
    text-indent: ${props => (props.type === 'text' ) && '30px'};
    font-size: ${props => (props.type === 'submit' || props.type === 'button') ? 'x-large' : 'medium' };
    padding: ${props => (props.type === 'submit' || props.type === 'button') ? '' : '15px 0' };

    `;

Input.displayName = 'Input'