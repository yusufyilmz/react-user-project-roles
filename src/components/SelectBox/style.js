

// import styled from 'styled-components';

// export const Ul = styled.ul`
// z-index: 10;
// position: absolute;
// width: 70%;
// border: 1px solid #dfdfdf;
//  border-bottom-right-radius: 3px; 
//  border-bottom-left-radius: 3px; 
// background-color: #fff;
// -webkit-box-shadow: 0 2px 5px -1px #e8e8e8;
// box-shadow: 0 2px 5px -1px #e8e8e8;
// font-weight: 700;
// max-height: 215px;
// overflow-y: scroll;
// -webkit-overflow-scrolling: touch;
//   `;

// export const Li = styled.li`
//     width: 100%;
//     font-size: 1rem;
//     padding: 8px 10px;
//     line-height: 1.6rem;
//     cursor: default;
//     display: inline-block;
//     white-space: nowrap;
//     text-overflow: ellipsis;
//   &.selected {
//     color: #fff;
//     background-color: #27435c;
//   }
//   &:hover {
//     color: white;
//     background-color: #27435c;
//   }`;






import styled from 'styled-components';

export const Ul = styled.ul`
    z-index: 10;
    position: absolute;
    width: 57%;
    margin-top: 0px;
    border: 1px solid #dfdfdf;
    border-bottom-right-radius: 3px;
    border-bottom-left-radius: 3px;
    background-color: #fff;
    -webkit-box-shadow: 0 2px 5px -1px #e8e8e8;
    box-shadow: 0 2px 5px -1px #e8e8e8;
    font-weight: 700;
    padding: 15px 0;
    max-height: 215px;
    overflow-y: scroll;
    -webkit-overflow-scrolling: touch;
    @media screen and (max-width: 600px) {
      width: 75%;
    }`;

export const Li = styled.li`
    width: 100%;
    font-size: 1rem;
    padding: 8px 10px;
    line-height: 1.6rem;
    cursor: default;
    display: inline-block;
    white-space: nowrap;
    text-overflow: ellipsis;
    &:selected {
      color: #fff;
      background-color: #27435c;
    }
    &:hover {
      color: black;
      background-color: #27435c;
    }`;
