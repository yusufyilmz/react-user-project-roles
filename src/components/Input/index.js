import React, { Fragment } from 'react';
import { Input, Label } from './style';


export default ({ type, name, text, onChange, value, onClick, required, placeHolder, width}) => {
    return (
        <Fragment>
            <Label htmlFor={name} >{text}</Label>
            <Input
                placeholder={placeHolder}
                width={width}
                name={name}
                autoComplete="off"
                required={required}
                id={name}
                type={type}
                onChange={onChange}
                onClick={onClick}
                value={value}>
            </Input>
        </Fragment>
    )
}


