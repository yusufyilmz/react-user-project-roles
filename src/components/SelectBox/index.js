import React, { Component } from 'react';
import { Ul, Li } from './style';
import Input from '../Input';
import onClickOutside from 'react-onclickoutside'


class SelectBox extends Component {

    constructor(props) {
        super(props);
        this.state = {
            listOpen: false,
            value: '',
            list: this.props.list,
        }
    }

    componentDidUpdate(prevProps) {
        if (prevProps.list !== this.props.list) {
            this.setState({ list: this.props.list })
        }
    }

    toggleList = (e) => {
        this.setState(prevState => ({
            listOpen: !prevState.listOpen,
            list: this.props.list,
            value: ''
        }))
    }

    selectItem(item) {
        this.setState(prevState => ({
            value: item.name,
            listOpen: !prevState.listOpen
        }), () => {
            this.props.selectItem(item);
        })
    }

    filterList = (e) => {
        const text = e.target.value;
        var list = this.props.list.filter(x => x.name.toLowerCase().startsWith(text.toLowerCase()))
        this.setState(prevState => ({
            value: text,
            list,
            listOpen: list.length > 0
        }), () => {
            this.props.selectItem({ name: text });

        })
    }
    handleClickOutside = (e) => {
        this.setState(prevState => ({
            listOpen: false
        }))
    }

    render() {
        return (
            <div>
                <Input
                    placeHolder={this.props.placeHolder}
                    text={this.props.text}
                    name={this.props.name}
                    onChange={this.filterList}
                    type="text"
                    required={true}
                    value={this.state.value}
                    onClick={this.toggleList} />
                {this.state.listOpen && <Ul >
                    {this.state.list.map((item) => {
                        if (item) return <Li
                            key={item.id}
                            onClick={() => this.selectItem(item)}>
                            {item.name}
                        </Li>
                    })}
                </Ul>
                }
            </div>
        )
    }

}

export default onClickOutside(SelectBox);
