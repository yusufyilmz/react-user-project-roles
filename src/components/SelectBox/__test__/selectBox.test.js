

import React from 'react';
import SelectBox from '../index';
import Input from '../../Input';
import { mount } from 'enzyme'


describe('SelectBox component', () => {


    let props, wrapper;

    beforeEach(() => {
        props = {
            name: 'testName',
            text: 'testText',
            placeHolder: 'testPlaceholder',
            selectItem : jest.fn(),
            list: [{
                id: 1,
                name: 'test'
              }
            ]
        }
        wrapper = mount(
            <SelectBox
                {...props}
                required={true} />
        );
    });

    test('renders element correctly', () => {
        expect(wrapper.find(Input).length).toBe(1)

    });

    test('renders element prop correctly', () => {
        expect(wrapper.find('Input').prop('placeholder')).toBe(props.placeHolder);
    });


    test('renders click correctly', () => {
        wrapper.find('Input').simulate('click');
        expect(wrapper.find('Ul')).toBeTruthy()
    });


    test('renders click and new elements correctly', () => {
        wrapper.find('Input').simulate('click');
        expect(wrapper.find('li').length).toBe(props.list.length)
    });


    test('renders select item correctly', () => {
        wrapper.find('Input').simulate('click');
        // let li = wrapper.find('li')[0]
        wrapper.find('li').simulate('click');
        expect(props.selectItem.mock.calls.length).toEqual(1);
    });


    test('renders select item and value correctly', () => {
        wrapper.find('Input').simulate('click');
        wrapper.find('li').simulate('click');
        expect(wrapper.find('Input').props()['value']).toBe(props.list[0].name);
    });


    test('renders element correctly', () => {
        // expect(wrapper).toMatchSnapshot();
    });

})
