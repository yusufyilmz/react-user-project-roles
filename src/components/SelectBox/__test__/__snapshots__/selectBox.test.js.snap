// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox component snapchat correctly 1`] = `
<OnClickOutside(SelectBox)
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  list={
    Array [
      Object {
        "id": 1,
        "name": "test",
      },
    ]
  }
  name="testName"
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  placeHolder="testPlaceholder"
  preventDefault={false}
  required={true}
  selectItem={[MockFunction]}
  stopPropagation={false}
  text="testText"
>
  <SelectBox
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    list={
      Array [
        Object {
          "id": 1,
          "name": "test",
        },
      ]
    }
    name="testName"
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    placeHolder="testPlaceholder"
    preventDefault={false}
    required={true}
    selectItem={[MockFunction]}
    stopPropagation={false}
    text="testText"
  >
    <div>
      <Component
        name="testName"
        onChange={[Function]}
        onClick={[Function]}
        placeHolder="testPlaceholder"
        required={true}
        text="testText"
        type="text"
        value=""
      >
        <Label
          htmlFor="testName"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "hApbuX",
                  "rules": Array [
                    "
    float: left;
    width: 25%;
    margin: 15px 5px;",
                  ],
                },
                "displayName": "Label",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "label",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            htmlFor="testName"
          >
            <label
              className="sc-htpNat hApbuX"
              htmlFor="testName"
            >
              testText
            </label>
          </StyledComponent>
        </Label>
        <Input
          autoComplete="off"
          id="testName"
          name="testName"
          onChange={[Function]}
          onClick={[Function]}
          placeholder="testPlaceholder"
          required={true}
          type="text"
          value=""
        >
          <StyledComponent
            autoComplete="off"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "beFuQG",
                  "rules": Array [
                    "
    margin-left: auto;
    margin-right: auto;
    width: 95%;
    padding: 12px;
    border: 1px solid #ccc;
    border-radius: 4px;
    resize: vertical;
    background-color: ",
                    [Function],
                    "
    display: ",
                    [Function],
                    "
    justify-content: ",
                    [Function],
                    "
    color: ",
                    [Function],
                    "
    ",
                  ],
                },
                "displayName": "Input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            id="testName"
            name="testName"
            onChange={[Function]}
            onClick={[Function]}
            placeholder="testPlaceholder"
            required={true}
            type="text"
            value=""
          >
            <input
              autoComplete="off"
              className="sc-bxivhb beFuQG"
              id="testName"
              name="testName"
              onChange={[Function]}
              onClick={[Function]}
              placeholder="testPlaceholder"
              required={true}
              type="text"
              value=""
            />
          </StyledComponent>
        </Input>
      </Component>
    </div>
  </SelectBox>
</OnClickOutside(SelectBox)>
`;
